src/main/java/org/openstreetmap/atlas/checks/distributed/IntegrityCheckSparkJob.java

```diff
@@ -93,12 +95,15 @@ public static void main(final String[] args)
      * @param configuration
      *            {@link MapRouletteConfiguration} to create a new {@link MapRouletteClient}s
      */
-    @SuppressWarnings("rawtypes")
     private static void executeChecks(final String country, final Atlas atlas,
-            final Set<BaseCheck> checksToRun, final MapRouletteConfiguration configuration)
+            final Set<BaseCheck<?>> checksToRun, final ExternalDataFetcher fetcher,
+            final MapRouletteConfiguration configuration)
     {
         final Pool checkExecutionPool = new Pool(checksToRun.size(), "Check execution pool",
                 POOL_DURATION_BEFORE_KILL);
+        checksToRun.stream().filter(ExternalDataUser.class::isInstance)
```

Comment:
My initial implementation actually did use a constructor. IIRC, I had some issues with serialization and deserialization in my runs, along with some odd issues with sparkcontexts. I'll revisit this. And since you mention it, doing the fetching in a threaded manner would probably speed things up a bit.

Reply:
I've rewritten the logic to use the constructor. I've now tested it with an HDFS system with multiple nodes, so I'm keeping the constructor logic.